app/scripts/lib/setupSentry.js

```diff
@@ -510,53 +509,70 @@ export default function setupSentry({ release, getState }) {
   });
 
   /**
-   * As long as a reference to the Sentry Hub can be found, and the user has
-   * opted into MetaMetrics, change the autoSessionTracking option and start
-   * a new sentry session.
+   * As long as a reference to the Sentry Hub can be found, and the session has
+   * not been started, start a new sentry session.
    */
   const startSession = async () => {
-    const hub = Sentry.getCurrentHub?.();
-    const options = hub.getClient?.().getOptions?.() ?? {};
-    if (hub && (await getMetaMetricsEnabled()) === true) {
-      options.autoSessionTracking = true;
-      hub.startSession();
+    const isHubDefined = Boolean(Sentry.getCurrentHub());
+    const { isSessionStarted } = global.sentry;
+    const isMetaMetricsEnabled = await getMetaMetricsEnabled();
+
+    if (
+      isHubDefined &&
+      (isSessionStarted === undefined || isSessionStarted === false) &&
+      isMetaMetricsEnabled === true
+    ) {
+      try {
+        Sentry.getCurrentHub().startSession();
+      } finally {
+        Sentry.getCurrentHub().captureSession();
+      }
+
+      global.sentry.isSessionStarted = true;
     }
   };
 
   /**
-   * As long as a reference to the Sentry Hub can be found, and the user has
-   * opted out of MetaMetrics, change the autoSessionTracking option and end
-   * the current sentry session.
+   * As long as a reference to the Sentry Hub can be found, and the session has
+   * already been started, end the current sentry session.
    */
   const endSession = async () => {
-    const hub = Sentry.getCurrentHub?.();
-    const options = hub.getClient?.().getOptions?.() ?? {};
-    if (hub && (await getMetaMetricsEnabled()) === false) {
-      options.autoSessionTracking = false;
-      hub.endSession();
+    const isHubDefined = Boolean(Sentry.getCurrentHub());
+    const { isSessionStarted } = global.sentry;
+    const isMetaMetricsEnabled = await getMetaMetricsEnabled();
+
+    if (
+      isHubDefined &&
+      isSessionStarted === true &&
+      isMetaMetricsEnabled === false
```

Comment:
If `isMetaMetricsEnabled === false` then we should not capture any session, right?

Reply:
`captureSession(true)` sends a request that ends a previously started session.
> Optional endSession: boolean
If set the session will be marked as exited and removed from the scope



https://getsentry.github.io/sentry-javascript/classes/_sentry_browser.Hub.html#captureSession